src/ol_openedx_canvas_integration/client.py

```diff
@@ -0,0 +1,204 @@
+        ))
+        return self._paginate(url)
+
+    def get_assignments_by_int_id(self):
+        assignments = self.list_canvas_assignments()
+        assignments_dict = {
+            assignment.get("integration_id"): assignment["id"]
+            for assignment in assignments
+            if assignment.get("integration_id") is not None
+        }
+        assignments_without_integration_id = sorted([
+            assignment["id"] for assignment in assignments if assignment.get("integration_id") is None
+        ])
+        if assignments_without_integration_id:
+            log.warning(
+                "These assignments are missing an integration_id: %s",
+                ", ".join(str(assignment_id) for assignment_id in assignments_without_integration_id)
+            )
+        return assignments_dict
+
+    def list_canvas_grades(self, assignment_id):
+        """
+        List grades for a Canvas assignment
+
+        Args:
+            assignment_id (int): The canvas assignment id
+        """
+        url = urljoin(
+            settings.CANVAS_BASE_URL,
+            "/api/v1/courses/{course_id}/assignments/{assignment_id}/submissions".format(
+                course_id=self.canvas_course_id,
+                assignment_id=assignment_id,
+            )
+        )
+        return self._paginate(url)
+
+    def create_canvas_assignment(self, payload):
+        """
+        Create an assignment on Canvas
+
+        Args:
+            payload (dict):
+        """
+        return self.session.post(
+            url=urljoin(
+                settings.CANVAS_BASE_URL,
+                "/api/v1/courses/{course_id}/assignments".format(course_id=self.canvas_course_id)
+            ),
+            json=payload,
+        )
+
+    def update_assignment_grades(self, canvas_assignment_id, payload):
+        return self.session.post(
+            url=urljoin(
+                settings.CANVAS_BASE_URL,
+                "/api/v1/courses/{course_id}/assignments/{assignment_id}/submissions/update_grades".format(
+                    course_id=self.canvas_course_id,
+                    assignment_id=canvas_assignment_id
+                )
+            ),
+            data=payload,
+        )
+
+
+def create_assignment_payload(subsection_block):
+    """
+    Create a Canvas assignment dict matching a subsection block on edX
+
+    Args:
+        subsection_block (openedx.core.djangoapps.content.block_structure.block_structure.BlockData):
+            The block data for the graded assignment/exam (in the structure of a course, this unit is a subsection)
+
+    Returns:
+        dict:
+            Assignment payload to be sent to Canvas to create or update the assignment
+    """
+    return {
+        "assignment": {
+            "name": subsection_block.display_name,
+            "integration_id": str(subsection_block.location),
+            "grading_type": "percent",
+            "points_possible": DEFAULT_ASSIGNMENT_POINTS,
```

Comment:
possible grade of actual subsection instead of default value.

Reply:
This would be done as part of https://github.com/mitodl/open-edx-plugins/issues/10 since it will be easier to directly get the max weightage/score for a subsection.